### PR TITLE
[WIP] feat: User and password management pages

### PR DESCRIPTION
**Pull request by TanoVip123**
Add 2 pages for webapp
password reset page 
user profile page
might need restyling

### CODE REVIEW

**Review by JZ-FSDev — APPROVED** [review]

File: webapp/package.json [hunk 1]
```diff
@@ -22,19 +22,28 @@
...
     "globals": "^15.9.0",
+    "happy-dom": "^15.7.4",
     "nyc": "^17.0.0",
     "playwright-test-coverage": "^1.2.12",
     "prettier": "^3.3.3",
     "prettier-plugin-svelte": "^3.2.6",
+    "prettier-plugin-tailwindcss": "^0.6.5",
     "svelte": "^4.2.19",
     "svelte-check": "^4.0.2",
+    "tailwindcss": "^3.4.9",
```
Comment (mayallhs) [inline webapp/package.json:46]: We can remove the tailwind dependencies if we are not using it!

Comment (TanoVip123) [reply webapp/package.json:46]: yeah will do

File: webapp/src/app.css [hunk 2]
```diff
@@ -78,6 +80,50 @@ pre {
...
+    cursor: pointer;
+    transition: background-color 0.3s;
+}
+
+.auth-form {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-items: center;
+    width: auto;
```
Comment (mayallhs) [inline webapp/src/app.css:109]: Default values for height and width are auto so these aren't needed to be explicitly written.

File: webapp/src/routes/auth/signup/+page.svelte [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                    successMessage =
+                        'Account created successfully. Normally we would ask for email verification but for demo... nah';
+                    accountCreated = true;
+                    errorMessage = '';
+                } else {
+                    const data = await response.json();
+                    errorMessage = data.errors[0].message;
+                }
+            } catch {
+                errorMessage = 'Something went wrong in the server T_T';
```
Comment (mayallhs) [inline webapp/src/routes/auth/signup/+page.svelte:53]: We would be changing these error messages later right?

Comment (TanoVip123) [reply webapp/src/routes/auth/signup/+page.svelte:53]: Yes lol

**Review by mayallhs — APPROVED** [review]

**Pull request merged**